src/coprocessor/codec/mysql/decimal.rs

```diff
@@ -1606,19 +1617,34 @@ impl Decimal {
         if inner_idx != 0 {
             d.word_buf[word_idx] = word * TEN_POW[DIGITS_PER_WORD as usize - inner_idx];
         }
-
-        if end_idx + 1 < bs.len() && (bs[end_idx] == b'e' || bs[end_idx] == b'E') {
+        if end_idx < bs.len() && (bs[end_idx] == b'e' || bs[end_idx] == b'E') {
             let exp = convert::bytes_to_int_without_context(&bs[end_idx + 1..])?;
             if exp > i64::from(i32::MAX) / 2 {
-                d.reset_to_zero();
-                return Ok(Res::Overflow(d.unwrap()));
+                return Ok(Res::Overflow(max_or_min_dec(
```

Comment:
Why not reuse the decimal?

Reply:
seems we should define a function similar to `max_decimal`?